src/main/java/org/jenkinsci/plugins/github/internal/GitHubLoginFunction.java

```diff
@@ -101,16 +99,13 @@ private Proxy getProxy(String apiUrl) {
      * Uses proxy of jenkins
      * If cache size > 0, uses cache
      *
-     * @param apiUrl to build proxy
-     *
      * @return connector to be used as backend for client
      */
-    private OkHttpConnector connector(String apiUrl) {
-        OkHttpClient client = new OkHttpClient().setProxy(getProxy(apiUrl));
+    private OkHttpConnector connector(GitHubServerConfig config) {
+        OkHttpClient client = new OkHttpClient().setProxy(getProxy(defaultIfBlank(config.getApiUrl(), GITHUB_URL)));
```

Comment:
wouldn't this mask issue when something is broken and url get blank?
i.e. user entered url, jelly magic failed, connection created with public connection and user didn't get thrown error